### ISSUE TITLE
Upgrade Akka

### ISSUE THREAD

**Issue by DanSimon** [tumblr/colossus#372]
Right now we're on Akka 2.3.9, we should upgrade to the latest, probably 2.4.3

**Comment by hntd187** [comment]
It should be important to note akka 2.4.X is Scala 2.11+ and Java8+

**Comment by DanSimon** [comment]
Ah that's a good point.  I suspect then this will get pushed off a bit.  How soon probably depends if we can easily do builds for both 2.3.x and 2.4.x

The good news is I did the upgrade myself just for fun and everything seemed to work fine.  The same goes for using Scala 2.12, with the exception that there appears to be no artifact of akka-agent published right now.  So we'll likely do both at the same time, probably in some kind of experimental branch.

**Comment by hntd187** [comment]
I believe as long as your language dependencies are there it should be fine, the migration guide says it's gucci

http://doc.akka.io/docs/akka/2.4.1/project/migration-guide-2.3.x-2.4.x.html

**Comment by hntd187** [comment]
Hey @DanSimon just thought I'd let you know, in some internal stuff I am working on I evicted akka 2.3 in favor of 2.4.9 and it's been running perfectly fine for awhile. So to test this further, I forked colossus and upgraded the dependency and other than some mild deprecation warnings all the tests passed for me. Just figured I'd let you know when you decide to upgrade or if you want to push another artifact the upgrade for it seems rather painless at face value.
